UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs

```
@@ -386,7 +378,6 @@ public void SetPaletteOfCurrentSprite(List<Color> newPalette, bool networked = t
 
 	public void PushTexture(bool networked = true)
 	{
-		if (initialised == false) TryInit();
```

I think the problem is there's a few places Where Sprite hand function is called  in awake in a different script, potentially leading to scenario where It's called before being initialised

---

What's the outcome if this occurs? An error? Or fails silently / unexpected results?

---

An error /unexpected results
since it was called before it would then override the Sprite with the one default set
 and it possibly could NRE on the spriteRenderer reference

---

Alright. I'd be happy to merge if it threw errors as they can be easily found, but it sounds like it'll create issues that might take a while to notice.